Sources/ContainerClient/Parser.swift

```diff
@@ -354,17 +354,40 @@ public struct Parser {
         case 1:
             throw ContainerizationError(.invalidArgument, message: "anonymous volumes are not supported")
         case 2, 3:
-            // Bind / volume mounts.
             let src = String(parts[0])
             let dst = String(parts[1])
 
-            let abs = URL(filePath: src).absoluteURL.path
-            if !FileManager.default.fileExists(atPath: abs) {
-                throw ContainerizationError(.invalidArgument, message: "named volumes are not supported")
+            var sourcePath = src
+
+            // Check if it's an absolute directory path first
+            if src.hasPrefix("/") {
+                let url = URL(filePath: src)
+                let absolutePath = url.absoluteURL.path
+
+                var isDirectory: ObjCBool = false
+                guard FileManager.default.fileExists(atPath: absolutePath, isDirectory: &isDirectory) else {
+                    throw ContainerizationError(.invalidArgument, message: "path '\(src)' does not exist")
+                }
+                guard isDirectory.boolValue else {
+                    throw ContainerizationError(.invalidArgument, message: "path '\(src)' is not a directory")
+                }
+                sourcePath = absolutePath
+            } else {
+                // Must be a named volume - validate name and check if it exists
+                guard VolumeStorage.isValidVolumeName(src) else {
+                    throw ContainerizationError(.invalidArgument, message: "invalid volume name '\(src)': only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed")
+                }
+
+                do {
+                    let response = try await ClientVolume.inspect(src)
+                    sourcePath = response.volume.mountpoint
```

Comment:
I'm not sure I understand this bit. It looks like we return a `Filesystem.virtiofs` regardless of whether we are creating a named volume mount or a virtiofs shared mount?

The other thing is that I'd expect a Parser not to be `async`. The parser should perform syntactic validation and parsing, and semantic validation (that a volume is there) should be performed on the parsed result.